gateway-api: Bump version to v0.6.0

sayboras opened the pull request:
## Description

This PR is to bump gateway-api version to v0.6.0, which includes the below changes:

- Change API version for ReferenceGrant to v1beta1
- Rename HTTPRequestHeaderFilter to HTTPHeaderFilter
- Update Gateway status to Programmed and Accepted
- Add support for http redirect filter
- Add support for response header filter
- Add validation for RouteDestinationPortMatching
- Update ConformanceTest to follow the upstream structure

Full changelog can be found in the below link https://github.com/kubernetes-sigs/gateway-api/releases/tag/v0.6.0

## Testing

Testing was done by running the whole conformance suite with below command. 

```
$ go test -v ./operator/pkg/gateway-api --gateway-class cilium --supported-features ReferenceGrant,TLSRoute,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,HTTPResponseHeaderModification,RouteDestinationPortMatching,GatewayClassObservedGenerationBump  -test.run "TestConformance"
```

![image](https://user-images.githubusercontent.com/9019229/220343613-7774f526-5875-4f92-8875-7ee4300a09c6.png)


<details>
<summary>Running conformance tests with all features</summary>

```
$ go test -v ./operator/pkg/gateway-api --gateway-class cilium --supported-features ReferenceGrant,TLSRoute,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,HTTPResponseHeaderModification,RouteDestinationPortMatching,GatewayClassObservedGenerationBump -test.run "TestConformance"
--- PASS: TestConformance (42.82s)
    --- PASS: TestConformance/GatewayInvalidRouteKind (1.08s)
        --- PASS: TestConformance/GatewayInvalidRouteKind/Gateway_listener_should_have_a_false_ResolvedRefs_condition_with_reason_InvalidRouteKinds_and_no_supportedKinds (1.01s)
        --- PASS: TestConformance/GatewayInvalidRouteKind/Gateway_listener_should_have_a_false_ResolvedRefs_condition_with_reason_InvalidRouteKinds_and_HTTPRoute_must_be_put_in_the_supportedKinds (0.00s)
    --- PASS: TestConformance/GatewayInvalidTLSConfiguration (0.15s)
        --- PASS: TestConformance/GatewayInvalidTLSConfiguration/Nonexistent_secret_referenced_as_CertificateRef_in_a_Gateway_listener (0.01s)
        --- PASS: TestConformance/GatewayInvalidTLSConfiguration/Malformed_secret_referenced_as_CertificateRef_in_a_Gateway_listener (1.01s)
        --- PASS: TestConformance/GatewayInvalidTLSConfiguration/Unsupported_kind_resource_referenced_as_CertificateRef_in_a_Gateway_listener (1.01s)
        --- PASS: TestConformance/GatewayInvalidTLSConfiguration/Unsupported_group_resource_referenced_as_CertificateRef_in_a_Gateway_listener (1.01s)
    --- PASS: TestConformance/GatewaySecretInvalidReferenceGrant (0.21s)
        --- PASS: TestConformance/GatewaySecretInvalidReferenceGrant/Gateway_listener_should_have_a_false_ResolvedRefs_condition_with_reason_RefNotPermitted (0.01s)
    --- PASS: TestConformance/GatewaySecretMissingReferenceGrant (1.04s)
        --- PASS: TestConformance/GatewaySecretMissingReferenceGrant/Gateway_listener_should_have_a_false_ResolvedRefs_condition_with_reason_RefNotPermitted (1.02s)
    --- PASS: TestConformance/GatewaySecretReferenceGrantAllInNamespace (1.08s)
        --- PASS: TestConformance/GatewaySecretReferenceGrantAllInNamespace/Gateway_listener_should_have_a_true_ResolvedRefs_condition_and_a_true_Programmed_condition (1.01s)
    --- PASS: TestConformance/GatewaySecretReferenceGrantSpecific (1.07s)
        --- PASS: TestConformance/GatewaySecretReferenceGrantSpecific/Gateway_listener_should_have_a_true_ResolvedRefs_condition_and_a_true_Programmed_condition (1.01s)
    --- PASS: TestConformance/HTTPRouteCrossNamespace (1.04s)
        --- PASS: TestConformance/HTTPRouteCrossNamespace/Simple_HTTP_request_should_reach_web-backend (1.01s)
    --- PASS: TestConformance/HTTPRouteDisallowedKind (1.09s)
        --- PASS: TestConformance/HTTPRouteDisallowedKind/Route_should_not_have_Parents_set_in_status (0.00s)
        --- PASS: TestConformance/HTTPRouteDisallowedKind/Gateway_should_have_0_Routes_attached (0.00s)
    --- PASS: TestConformance/HTTPExactPathMatching (0.04s)
        --- PASS: TestConformance/HTTPExactPathMatching/3_request_to_'/one/example'_should_receive_a_404 (1.00s)
        --- PASS: TestConformance/HTTPExactPathMatching/4_request_to_'/two/'_should_receive_a_404 (1.00s)
        --- PASS: TestConformance/HTTPExactPathMatching/2_request_to_'/'_should_receive_a_404 (1.00s)
        --- PASS: TestConformance/HTTPExactPathMatching/0_request_to_'/one'_should_go_to_infra-backend-v1 (1.01s)
        --- PASS: TestConformance/HTTPExactPathMatching/1_request_to_'/two'_should_go_to_infra-backend-v2 (1.01s)
    --- PASS: TestConformance/HTTPRouteHeaderMatching (0.04s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/5_request_to_'/'_with_headers_should_receive_a_404 (0.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/4_request_to_'/'_with_headers_should_receive_a_404 (0.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/10_request_to_'/'_with_headers_should_receive_a_404 (0.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/0_request_to_'/'_with_headers_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/1_request_to_'/'_with_headers_should_go_to_infra-backend-v2 (1.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/9_request_to_'/'_with_headers_should_go_to_infra-backend-v2 (1.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/3_request_to_'/'_with_headers_should_go_to_infra-backend-v2 (1.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/2_request_to_'/'_with_headers_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/7_request_to_'/'_with_headers_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/6_request_to_'/'_with_headers_should_go_to_infra-backend-v1 (1.01s)
        --- PASS: TestConformance/HTTPRouteHeaderMatching/8_request_to_'/'_with_headers_should_go_to_infra-backend-v2 (1.00s)
    --- PASS: TestConformance/HTTPRouteHostnameIntersection (5.21s)
        --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames (5.02s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/11_request_to_'foo.wildcard.io/non-matching-prefix'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/1_request_to_'non.matching.com/s1'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/3_request_to_'foo.wildcard.io/s1'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/13_request_to_'non.matching.com/s3'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/22_request_to_'very.specific.com/s4'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/23_request_to_'foo.anotherwildcard.io/non-matching-prefix'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/14_request_to_'foo.specific.com/s3'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/16_request_to_'very.specific.com/non-matching-prefix'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/8_request_to_'non.matching.com/s2'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/20_request_to_'anotherwildcard.io/s4'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/21_request_to_'foo.wildcard.io/s4'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/10_request_to_'very.specific.com/s2'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/4_request_to_'very.specific.com/non-matching-prefix'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/9_request_to_'wildcard.io/s2'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/2_request_to_'foo.nonmatchingwildcard.io/s1'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/15_request_to_'foo.wildcard.io/s3'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/19_request_to_'foo.bar.anotherwildcard.io/s4'_should_go_to_infra-backend-v1 (0.01s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/18_request_to_'bar.anotherwildcard.io/s4'_should_go_to_infra-backend-v1 (0.01s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/17_request_to_'foo.anotherwildcard.io/s4'_should_go_to_infra-backend-v1 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/0_request_to_'very.specific.com/s1'_should_go_to_infra-backend-v1 (0.01s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/6_request_to_'bar.wildcard.io/s2'_should_go_to_infra-backend-v2 (0.01s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/5_request_to_'foo.wildcard.io/s2'_should_go_to_infra-backend-v2 (0.01s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/7_request_to_'foo.bar.wildcard.io/s2'_should_go_to_infra-backend-v2 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/12_request_to_'very.specific.com/s3'_should_go_to_infra-backend-v3 (0.01s)
        --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_not_intersect_with_listener_hostnames (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_not_intersect_with_listener_hostnames/0_request_to_'specific.but.wrong.com/s5'_should_receive_a_404 (0.00s)
            --- PASS: TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_not_intersect_with_listener_hostnames/1_request_to_'wildcard.io/s5'_should_receive_a_404 (0.00s)
    --- PASS: TestConformance/HTTPRouteInvalidNonExistentBackendRef (1.07s)
        --- PASS: TestConformance/HTTPRouteInvalidNonExistentBackendRef/HTTPRoute_with_only_a_nonexistent_BackendRef_has_a_ResolvedRefs_Condition_with_status_False_and_Reason_BackendNotFound (0.03s)
        --- PASS: TestConformance/HTTPRouteInvalidNonExistentBackendRef/HTTP_Request_to_invalid_nonexistent_backend_receive_a_500 (1.00s)
    --- PASS: TestConformance/HTTPRouteInvalidBackendRefUnknownKind (0.05s)
        --- PASS: TestConformance/HTTPRouteInvalidBackendRefUnknownKind/HTTPRoute_with_Invalid_Kind_has_a_ResolvedRefs_Condition_with_status_False_and_Reason_InvalidKind (0.00s)
        --- PASS: TestConformance/HTTPRouteInvalidBackendRefUnknownKind/HTTP_Request_to_invalid_backend_with_invalid_Kind_receives_a_500 (0.00s)
    --- PASS: TestConformance/HTTPRouteInvalidCrossNamespaceBackendRef (0.04s)
        --- PASS: TestConformance/HTTPRouteInvalidCrossNamespaceBackendRef/HTTPRoute_with_a_cross-namespace_BackendRef_and_no_ReferenceGrant_has_a_ResolvedRefs_Condition_with_status_False_and_Reason_RefNotPermitted (0.00s)
        --- PASS: TestConformance/HTTPRouteInvalidCrossNamespaceBackendRef/HTTP_Request_to_invalid_cross-namespace_backend_must_receive_a_500 (0.00s)
    --- PASS: TestConformance/HTTPRouteInvalidCrossNamespaceParentRef (0.03s)
        --- PASS: TestConformance/HTTPRouteInvalidCrossNamespaceParentRef/Route_should_not_have_Parents_set_in_status (0.01s)
        --- PASS: TestConformance/HTTPRouteInvalidCrossNamespaceParentRef/Gateway_should_have_0_Routes_attached (0.01s)
    --- PASS: TestConformance/HTTPRouteInvalidParentRefNotMatchingListenerPort (1.03s)
        --- PASS: TestConformance/HTTPRouteInvalidParentRefNotMatchingListenerPort/HTTPRoute_with_no_matching_port_in_ParentRef_has_an_Accepted_Condition_with_status_False_and_Reason_NoMatchingParent (1.01s)
        --- PASS: TestConformance/HTTPRouteInvalidParentRefNotMatchingListenerPort/Route_should_not_have_Parents_accepted_in_status (0.00s)
        --- PASS: TestConformance/HTTPRouteInvalidParentRefNotMatchingListenerPort/Gateway_should_have_0_Routes_attached (0.00s)
    --- PASS: TestConformance/HTTPRouteListenerHostnameMatching (3.15s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/6_request_to_'foo.com/'_should_receive_a_404 (0.00s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/7_request_to_'no.matching.host/'_should_receive_a_404 (0.00s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/2_request_to_'baz.bar.com/'_should_go_to_infra-backend-v3 (0.00s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/5_request_to_'multiple.prefixes.foo.com/'_should_go_to_infra-backend-v3 (0.00s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/3_request_to_'boo.bar.com/'_should_go_to_infra-backend-v3 (0.00s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/1_request_to_'foo.bar.com/'_should_go_to_infra-backend-v2 (0.00s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/4_request_to_'multiple.prefixes.bar.com/'_should_go_to_infra-backend-v3 (0.00s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/0_request_to_'bar.com/'_should_go_to_infra-backend-v1 (0.00s)
    --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes (0.08s)
        --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes/2_request_to_'example.net/example'_should_go_to_infra-backend-v1 (0.00s)
        --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes/0_request_to_'example.com/'_should_go_to_infra-backend-v1 (0.00s)
        --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes/1_request_to_'example.com/example'_should_go_to_infra-backend-v1 (0.00s)
        --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes/5_request_to_'example.net/v2'_should_go_to_infra-backend-v1 (0.00s)
        --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes/3_request_to_'example.com/example'_with_headers_should_go_to_infra-backend-v1 (0.00s)
        --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes/7_request_to_'example.com/'_with_headers_should_go_to_infra-backend-v2 (1.00s)
        --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes/4_request_to_'example.com/v2'_should_go_to_infra-backend-v2 (1.00s)
        --- PASS: TestConformance/HTTPRouteMatchingAcrossRoutes/6_request_to_'example.com/v2/example'_should_go_to_infra-backend-v2 (1.01s)
    --- PASS: TestConformance/HTTPRouteMatching (0.03s)
        --- PASS: TestConformance/HTTPRouteMatching/0_request_to_'/'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteMatching/2_request_to_'/'_with_headers_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteMatching/1_request_to_'/example'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteMatching/4_request_to_'/v2/example'_should_go_to_infra-backend-v2 (1.00s)
        --- PASS: TestConformance/HTTPRouteMatching/5_request_to_'/'_with_headers_should_go_to_infra-backend-v2 (1.01s)
        --- PASS: TestConformance/HTTPRouteMatching/3_request_to_'/v2'_should_go_to_infra-backend-v2 (1.01s)
    --- PASS: TestConformance/HTTPRouteMethodMatching (0.03s)
        --- PASS: TestConformance/HTTPRouteMethodMatching/2_request_to_'/'_should_receive_a_404 (1.00s)
        --- PASS: TestConformance/HTTPRouteMethodMatching/0_request_to_'/'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteMethodMatching/1_request_to_'/'_should_go_to_infra-backend-v2 (1.00s)
    --- PASS: TestConformance/HTTPRoutePartiallyInvalidViaInvalidReferenceGrant (1.06s)
        --- PASS: TestConformance/HTTPRoutePartiallyInvalidViaInvalidReferenceGrant/HTTPRoute_with_BackendRef_in_another_namespace_and_no_ReferenceGrant_covering_the_Service_has_a_ResolvedRefs_Condition_with_status_False_and_Reason_RefNotPermitted (0.00s)
        --- PASS: TestConformance/HTTPRoutePartiallyInvalidViaInvalidReferenceGrant/HTTP_Request_to_invalid_backend_with_missing_referenceGrant_should_receive_a_500 (1.00s)
        --- PASS: TestConformance/HTTPRoutePartiallyInvalidViaInvalidReferenceGrant/HTTP_Request_to_valid_sibling_backend_should_succeed (0.00s)
    --- PASS: TestConformance/HTTPRouteQueryParamMatching (0.03s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/7_request_to_'/?animal=dog'_should_receive_a_404 (1.00s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/6_request_to_'/?color=blue'_should_receive_a_404 (1.00s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/9_request_to_'/'_should_receive_a_404 (1.00s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/8_request_to_'/?animal=whaledolphin'_should_receive_a_404 (1.00s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/1_request_to_'/?animal=dolphin'_should_go_to_infra-backend-v2 (1.00s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/3_request_to_'/?ANIMAL=Whale'_should_go_to_infra-backend-v3 (1.01s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/4_request_to_'/?animal=whale&otherparam=irrelevant'_should_go_to_infra-backend-v1 (1.01s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/2_request_to_'/?animal=dolphin&color=blue'_should_go_to_infra-backend-v3 (1.01s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/5_request_to_'/?animal=dolphin&color=yellow'_should_go_to_infra-backend-v2 (1.01s)
        --- PASS: TestConformance/HTTPRouteQueryParamMatching/0_request_to_'/?animal=whale'_should_go_to_infra-backend-v1 (1.01s)
    --- PASS: TestConformance/HTTPRouteReferenceGrant (1.09s)
        --- PASS: TestConformance/HTTPRouteReferenceGrant/Simple_HTTP_request_should_reach_web-backend (1.00s)
    --- PASS: TestConformance/HTTPRouteRequestHeaderModifier (0.04s)
        --- PASS: TestConformance/HTTPRouteRequestHeaderModifier/0_request_to_'/set'_with_headers_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteRequestHeaderModifier/2_request_to_'/add'_with_headers_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteRequestHeaderModifier/4_request_to_'/remove'_with_headers_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteRequestHeaderModifier/3_request_to_'/add'_with_headers_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteRequestHeaderModifier/5_request_to_'/multiple'_with_headers_should_go_to_infra-backend-v1 (1.01s)
        --- PASS: TestConformance/HTTPRouteRequestHeaderModifier/1_request_to_'/set'_with_headers_should_go_to_infra-backend-v1 (1.01s)
        --- PASS: TestConformance/HTTPRouteRequestHeaderModifier/6_request_to_'/case-insensitivity'_with_headers_should_go_to_infra-backend-v1 (1.01s)
    --- PASS: TestConformance/HTTPRouteResponseHeaderModifier (0.04s)
        --- PASS: TestConformance/HTTPRouteResponseHeaderModifier/0_request_to_'/set'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteResponseHeaderModifier/2_request_to_'/add'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteResponseHeaderModifier/1_request_to_'/set'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteResponseHeaderModifier/3_request_to_'/add'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteResponseHeaderModifier/4_request_to_'/remove'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteResponseHeaderModifier/6_request_to_'/case-insensitivity'_should_go_to_infra-backend-v1 (1.00s)
        --- PASS: TestConformance/HTTPRouteResponseHeaderModifier/5_request_to_'/multiple'_should_go_to_infra-backend-v1 (1.01s)
    --- PASS: TestConformance/HTTPRouteSimpleSameNamespace (1.04s)
        --- PASS: TestConformance/HTTPRouteSimpleSameNamespace/Simple_HTTP_request_should_reach_infra-backend (1.00s)
PASS
ok      github.com/cilium/cilium/operator/pkg/gateway-api       42.851s


``` 

</details>

sayboras committed:
gateway-api: Bump the version to v0.6.0

This commit is to bump gateway-api version to v0.6.0, and adjust the
implementation accordingly.

- Change API version for ReferenceGrant to v1beta1
- Rename HTTPRequestHeaderFilter to HTTPHeaderFilter
- Update Gateway status to Programmed and Accepted
- Update ConformanceTest to follow the upstream version

Full changelog can be found in the below link
https://github.com/kubernetes-sigs/gateway-api/releases/tag/v0.6.0

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
gateway-api: Add support for HTTPResponseHeaderModification

This commit is to make sure that we have the support for response header
filter. The conformance test is enabled accordingly.

Kindly note this Response header filter is still in experimental.

Relates: #21924
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
gateway-api: Add support for RouteDestinationPortMatching

This commit is to make sure that we have the support RouteDestinationPortMatching
feature. The conformance test is enabled accordingly.

Relates: https://github.com/kubernetes-sigs/gateway-api/pull/1591
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
gateway-api: Add support for Redirect filter

This commit is to make sure that we have the support HTTP Redirect filter
feature, which is enabled by default in conformance test.

Relates: #21924
Signed-off-by: Tam Mach <tam.mach@cilium.io>

youngnick commented on operator/pkg/gateway-api/gateway_reconcile.go:
```diff
@@ -60,7 +61,13 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
+
+	if err := r.setListenerStatus(ctx, gw); err != nil {
+		scopedLog.WithError(err).Error("Unable to set listener status")
+		setGatewayReady(gw, false, "Unable to set listener status")
```
This one should also be `Accepted` set to false, not `Ready`.

youngnick commented on operator/pkg/gateway-api/gateway_reconcile.go:
```diff
@@ -276,11 +277,44 @@ func (r *gatewayReconciler) setAddressStatus(ctx context.Context, gw *gatewayv1b
+				},
+			}
+		}
+		var conds []metav1.Condition
```
We may need to update this with a special append, or move it to a hash, with the Type as the key, because Conditions, although they are stored as a slice, are actually interpreted by the apiserver as a map with cond.Type as the key. Using an intermediate hash, and then swapping the values back into a slice before applying the status update, should make sure this works as expected.

sayboras replied:
good point, I have a merge utility methods, let me switch to it.

https://github.com/cilium/cilium/blob/d0a43aa05ee39b884ff9e87318e2a6aebd5ca98f/operator/pkg/gateway-api/status.go#L29

youngnick reviewed (commented):
LGTM overall, just a couple of small things to fix.

sayboras committed:
gha: Update Gateway API Conformance test

This commit is to update GHA Gateway API conformance test as per below:

- Use experimental CRDs mainly for running tests for all features.
- Remove -debug flag as it's too verbose.
- Limit number of concurrency test from default (e.g. 8) to 4 to minimize
  overlapping between the tests.
- Enable GatewayClassObservedGenerationBump feature.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

youngnick commented on operator/pkg/gateway-api/gateway_reconcile.go:
```diff
@@ -370,3 +393,27 @@ func isReferenceAllowed(ctx context.Context, c client.Client, gw *gatewayv1beta1
+	return nil
+}
+
+func isValidPemFormat(b []byte) bool {
```
Nit: Technically, a PEM containing more than one block is _valid_, but I don't know if we want to accept it. (If there's more than one block, `rest` will have the other blocks as well).

sayboras replied:
good point :100:

youngnick reviewed (approved):
LGTM with one small nit about isValidPem.

rolinh reviewed (approved):
Vendor changes lgtm.
